src/cart/crt_swim.c

```diff
@@ -414,17 +481,24 @@ int crt_swim_enable(struct crt_grp_priv *grp_priv, int crt_ctx_idx)
 		D_GOTO(out, rc = -DER_INVAL);
 	}
 
+	if (crt_ctx_idx < 0) {
+		D_ERROR("Invalid context index\n");
+		D_GOTO(out, rc = -DER_INVAL);
+	}
+
 	D_RWLOCK_WRLOCK(&csm->csm_rwlock);
 	if (csm->csm_crt_ctx_idx != crt_ctx_idx) {
-		rc = crt_unregister_progress_cb(crt_swim_progress_cb,
-						csm->csm_crt_ctx_idx,
-						csm->csm_ctx);
-		if (rc)
-			D_DEBUG(DB_TRACE, "crt_unregister_progress_cb() "
-				"failed: rc=%d\n", rc);
+		if (csm->csm_crt_ctx_idx != -1) {
+			rc = crt_unregister_progress_cb(crt_swim_progress_cb,
+							csm->csm_crt_ctx_idx,
+							NULL);
+			if (rc)
+				D_DEBUG(DB_TRACE, "crt_unregister_progress_cb()"
```

Comment:
This should be error, we want to know if something drastic happened that prevented unregister_progress_cb fails